src/base/dof_map_constraints.C

```diff
@@ -1183,29 +1183,33 @@ public:
      * input function \p f gives the arbitrary solution.
      */
 
+    // Check for a quick return in case there's no work to be done.
+    if (dirichlets.empty())
+      return;
```

Comment:
You could probably drop these lines.  In the unexpected dirichlets.empty() case that would take an extra branch (one for the empty loop over dirichlets but then a second to test ```!system```), but in the common case it'd save a branch, and of course it makes the code a couple lines shorter.

Reply:
That's assuming we want to consider "applying an empty set of dirichlet constraints" to be a bug, though.  Which is reasonable as long as we're just calling it internally, but I guess external usage might want to be more flexible.  So you could leave it.

Reply:
I can't drop these lines because I no longer `return` when a valid System isn't found, I now throw an error. So I need to distinguish between the valid cases where there are no DirichletBoundary objects (e.g. introduction_ex2 whose failure originally brought this to my attention) and cases where there are DirichletBoundary objects but they aren't set up correctly in one way or another.